packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -1950,17 +1950,94 @@ class WebpackCLI implements IWebpackCLI {
+      // @ts-expect-error TODO remove when extends property type is added to webpack
+      config.options.extends
+    ) {
+      const merge = await this.tryRequireThenImport<typeof webpackMerge>("webpack-merge");
+
+      let extendedConfig = {};
+
+      // @ts-expect-error TODO remove when extends property type is added to webpack
+      if (Array.isArray(config.options.extends)) {
+        const extendedConfigs: {
+          options: ConfigOptions | ConfigOptions[];
+          path: string;
+        }[] = await Promise.all(
+          // @ts-expect-error TODO remove when extends property type is added to webpack
+          config.options.extends.map((configPath: string) =>
+            loadConfigByPath(path.resolve(configPath), options.argv),
+          ),
+        );
+
+        extendedConfig = extendedConfigs.reduce((accumulator: object, config) => {
+          return merge(accumulator, config.options);
+        }, {});
+      } else {
+        // load the config from the extends option
+        // @ts-expect-error TODO remove when extends property type is added to webpack
+        extendedConfig = loadConfigByPath(path.resolve(config.options.extends), options.argv);
+      }
+
+      config.options = merge(extendedConfig, config.options);
+    } else if (
+      Array.isArray(config.options) &&
+      // @ts-expect-error TODO remove when extends property type is added to webpack
+      config.options.some((options) => options.extends)
+    ) {
+      const merge = await this.tryRequireThenImport<typeof webpackMerge>("webpack-merge");
+
+      for (let index = 0; index < config.options.length; index++) {
+        const configOptions = config.options[index];
+        // @ts-expect-error TODO remove when extends property type is added to webpack
+        if (configOptions.extends) {
+          let extendedConfig = {};
+
+          // @ts-expect-error TODO remove when extends property type is added to webpack
+          if (Array.isArray(configOptions.extends)) {
+            const extendedConfigs: {
+              options: ConfigOptions | ConfigOptions[];
+              path: string;
+            }[] = await Promise.all(
+              // @ts-expect-error TODO remove when extends property type is added to webpack
+              configOptions.extends.map((configPath: string) =>
+                loadConfigByPath(path.resolve(configPath), options.argv),
+              ),
+            );
+
+            extendedConfig = extendedConfigs.reduce((accumulator: object, config) => {
+              return merge(accumulator, config.options);
+            }, {});
+          } else {
+            // load the config from the extends option
+            // @ts-expect-error TODO remove when extends property type is added to webpack
+            extendedConfig = loadConfigByPath(path.resolve(configOptions.extends), options.argv);
+          }
+
+          config.options[index] = merge(extendedConfig, configOptions);
+        }
+      }
```

Comment:
Please move it to own function and just use `&& { extend: string[] }` to avoid `ts-except-error`